net/netcheck/netcheck.go

```diff
@@ -980,6 +1023,45 @@ func (c *Client) finishAndStoreReport(rs *reportState, dm *tailcfg.DERPMap) *Rep
 	return report
 }
 
+var generate204URL = envknob.StringOrDefault(
+	"TS_NETCHECK_GENERATE_204_URL",
+	"http://controlplane.tailscale.com/generate_204",
```

Comment:
looks like this generates a 404

But I'd prefer to not hard-code this URL.

Can it not use DERP servers from the DERP map instead?

Reply:
It's possible that I'm hitting a different server from you, but...
```
$ curl -v http://controlplane.tailscale.com/generate_204
*   Trying 18.157.173.201:80...
* Connected to controlplane.tailscale.com (18.157.173.201) port 80 (#0)
> GET /generate_204 HTTP/1.1
> Host: controlplane.tailscale.com
> User-Agent: curl/7.83.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 204 No Content
< Date: Mon, 12 Sep 2022 16:17:07 GMT
< 
* Connection #0 to host controlplane.tailscale.com left intact
```

This is currently served by trunkd, but I could add the `/generate_204` endpoint to the DERPs, yeah - easy enough to do, if you'd prefer.

Reply:
https://github.com/tailscale/tailscale/pull/5601

Reply:
Using DERPs means headscale needn't be hitting our servers

Reply:
Okay, re-pushed to use the DERP servers. I can't merge this PR until we've deployed #5601 to all the DERPs, though, since the current behaviour is "redirect to HTTPS" and that reports as "has a captive portal", unfortunately.

Reply:
And if people are running old DERPers? What will happen if we false positive detect a captive portal? Is the plan to turn off their Tailscale routes?

Reply:
False-positives don't do anything right now other than cause "full" netchecks to happen more frequently, and even then, only if there's no UDP access. I'm pretty okay with that for people running old custom DERPs, TBH.

Reply:
@bradfitz - We have now deployed all the DERPs with this route (with thanks to @DentonGentry for doing most of the work). This PR should be ready for review now.